internal/c/parts/video/font/font.cpp

```diff
@@ -1,350 +1,339 @@
-        FT_Done_Face(fonts[i].handle);
-        return 0;
-    }
-    fonts[i].default_pixel_height = default_pixel_height;
-
-    /*
-    static float m_height; m_height=((float)fonts[i].handle->size->metrics.height)/64.0;
-    static float m_up; m_up=((float)fonts[i].handle->size->metrics.ascender)/64.0;
-    static float m_down; m_down=-(((float)fonts[i].handle->size->metrics.descender)/64.0);
-    static float m_char_height; m_char_height=m_up+m_down;
-    static float m_h; m_h=default_pixel_height;
-    fonts[i].baseline= (m_h/m_height) * ((m_height-m_char_height)/2.0+m_up) ;
-    */
-    static float m_height;
-    m_height = ((float)fonts[i].handle->size->metrics.height) / 64.0;
-    static float m_up;
-    m_up = ((float)fonts[i].handle->size->metrics.ascender) / 64.0;
-    static float m_h;
-    m_h = default_pixel_height;
-    fonts[i].baseline = qbr((m_up / m_height) * m_h);
-
-    if (options & 16) {
-        // get the width of capital W
-        static uint32 cp;
-        cp = 87;
-        static uint8 *data1;
-        int32 w1, h1, pre_x, post_x;
-        FontRenderTextUTF32(i, &cp, 1, 1, &data1, &w1, &h1, &pre_x, &post_x);
-        fonts[i].monospace_width = w1;
-        free(data1);
-        fonts[i].monospace = 1;
-    } // monospace
-
-    // Note: DO NOT ADD NEW CONTENT HERE, ADD IT ABOVE MONOSPACE CHECK
-
-    fonts[i].in_use = 1;
-    return i;
-}
-
-void FontFree(int32 i) {
-    FT_Done_Face(fonts[i].handle);
-    free(fonts[i].ttf_data);
-    fonts[i].in_use = 0;
-}
-
-int32 FontRenderTextASCII(int32 i, uint8 *codepoint, int32 codepoints, int32 options, uint8 **out_data, int32 *out_x, int32 *out_y, int32 *out_x_pre_increment,
-                          int32 *out_x_post_increment) {
-    static uint32 *utf32_codepoint;
-    static int32 retval;
-    if (codepoints >= 1) {
-        utf32_codepoint = (uint32 *)malloc(codepoints * 4);
-        static int32 x;
-        for (x = 0; x < codepoints; x++) {
-            utf32_codepoint[x] = codepage437_to_unicode16[codepoint[x]];
-        }
-    }
-    retval = FontRenderTextUTF32(i, utf32_codepoint, codepoints, options, out_data, out_x, out_y, out_x_pre_increment, out_x_post_increment);
-    if (codepoints > 0)
-        free(utf32_codepoint);
-    return retval;
-}
-
-int32 FontWidth(int32 i) {
-    if (fonts[i].monospace)
-        return fonts[i].monospace_width;
-    return 0;
-}
-
-#endif
+//----------------------------------------------------------------------------------------------------------------------
+//  QB64-PE Font Library
+//  Powered by FreeType 2.4.12 (https://github.com/vinniefalco/FreeTypeAmalgam)
+//----------------------------------------------------------------------------------------------------------------------
+
+//[ ]what if no glyph index could be found and it returns 0??
+//[ ]FT_Done_Face
+//[ ]Check for memory leaks
+
+#include "font.h"
+#include "freetypeamalgam.h"
+#include "rounding.h"
+
+extern uint16_t codepage437_to_unicode16[];
```

Comment:
I think this `extern` should go in `font.h`. We also might consider moving the actual declaration into this file.

Reply:
That's a good idea.👍 I'll make this change in the next PR.